### PR TITLE
Add RPC for latest checkpoint

### PR DESCRIPTION
**Pull request by bhgomes**
Signed-off-by: Brandon H. Gomes <bhgomes@pm.me>

## Description

Adds an RPC method for querying the latest checkpoint. This is useful for getting a loading bar for synchronization.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functonality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

**Review by Kevingislason — DISMISSED** [review]

**Comment by stechu** [comment]
why a new RPC end-point? Should we just add the total number to the `pull_ledger_diff` (to reduce number of RPCs).

**Comment by stechu** [comment]
This PR requires merging #723  first.

**Review by stechu — DISMISSED** [review]

**Review by Kevingislason — DISMISSED** [review]

File: pallets/manta-pay/src/lib.rs [hunk 1]
```diff
@@ -557,10 +557,14 @@ pub mod pallet {
             let (more_receivers, receivers) =
                 Self::pull_receivers(*checkpoint.receiver_index, max_receivers);
             let (more_senders, senders) = Self::pull_senders(checkpoint.sender_index, max_senders);
+            let receivers_total = (0..=255)
+                .map(|i| ShardTrees::<T>::get(i).current_path.leaf_index as u128)
```
Comment (stechu) [inline pallets/manta-pay/src/lib.rs:568]: @bhgomes  do we need +1 here?

Comment (stechu) [reply pallets/manta-pay/src/lib.rs:568]: should not +1 here.

**Review by stechu — APPROVED** [review]

File: pallets/manta-pay/src/lib.rs [hunk 2]
```diff
@@ -557,10 +557,15 @@ pub mod pallet {
             let (more_receivers, receivers) =
                 Self::pull_receivers(*checkpoint.receiver_index, max_receivers);
             let (more_senders, senders) = Self::pull_senders(checkpoint.sender_index, max_senders);
+            let senders_receivers_total = (0..=255)
```
Comment (ghzlatarev) [inline pallets/manta-pay/src/lib.rs:567]: I still don't quite get why this is better, than just making a separate storage item to keep track of the utxo set size.

**Review by Garandor — APPROVED** [review]
got the same thought as @ghzlatarev  in https://github.com/Manta-Network/Manta/pull/712/files#r935325448 that it would be cheaper to bump the total number in storage on each TX and read a StorageValue instead of collecting the 256 shards on *every single rpc call*.

Intuitively, it sounds far more computationally expensive to have 257 storage reads per RPC instead of 2, especially in light of the recently merged #469 

I think this'll need proper benchmarking - or some caching in the #686 where it only pulls a new value from the RT once per block and serves the rest from cache.

If we do the latter thing this LGTM. ~~@bhgomes could you pls open a followup issue?~~ Nvm, just realized that the #686 indexer IS the followup issue

**Review by ghzlatarev — APPROVED** [review]

**Review by Dengjianping — APPROVED** [review]

**Comment by 0xduracell** [comment]
> got the same thought as @ghzlatarev in https://github.com/Manta-Network/Manta/pull/712/files#r935325448 that it would be cheaper to bump the total number in storage on each TX and read a StorageValue instead of collecting the 256 shards on _every single rpc call_.
> 
> Intuitively, it sounds far more computationally expensive to have 257 storage reads per RPC instead of 2, especially in light of the recently merged #469
> 
> I think this'll need proper benchmarking - or some caching in the #686 where it only pulls a new value from the RT once per block and serves the rest from cache.
> 
> If we do the latter thing this LGTM. ~@bhgomes could you pls open a followup issue?~ Nvm, just realized that the #686 indexer IS the followup issue

Yeah, we have discussed this. Will do it in the `MantaPay 1.0.0`.

**Pull request merged**